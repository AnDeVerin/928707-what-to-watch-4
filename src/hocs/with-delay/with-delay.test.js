import React from 'react';
import renderer from 'react-test-renderer';
import withDelay from './with-delay.js';

const MockComponent = (props) => <div {...props} />;

const MockComponentWrapped = withDelay(MockComponent);

const movie = {
  id: 1,
  title: 'Aviator',
  thumbUrl: 'img/aviator.jpg',
  trailer: 'https://example.com/video.mp4',
};

it(`withDelay is rendered correctly`, () => {
  const tree = renderer
    .create(<MockComponentWrapped movie={movie} onClick={jest.fn()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
