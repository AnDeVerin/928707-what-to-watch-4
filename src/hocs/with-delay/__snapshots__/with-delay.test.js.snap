// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDelay is rendered correctly 1`] = `
<div
  isPlaying={false}
  movie={
    Object {
      "thumbUrl": "img/aviator.jpg",
      "title": "Aviator",
      "trailer": "https://example.com/video.mp4",
    }
  }
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
